File: backend/app/schemas/stats.py
from datetime import date

from pydantic import BaseModel


class DailyCount(BaseModel):
    day: date
    attempts: int
    correct: int


class CategoryAccuracy(BaseModel):
    category: str
    attempts: int
    correct: int
    accuracy: float


class DashboardStats(BaseModel):
    total_questions: int
    total_attempts: int
    overall_accuracy: float
    streak_days: int
    due_today: int
    daily_7d: list[DailyCount]
    weak_categories: list[CategoryAccuracy]


class HeatmapCell(BaseModel):
    category: str
    difficulty: int
    attempts: int
    correct: int
    accuracy: float


class HeatmapResponse(BaseModel):
    cells: list[HeatmapCell]

File: frontend/app/lib/stats.ts
export type DailyCount = {
  day: string;
  attempts: number;
  correct: number;
};

export type CategoryAccuracy = {
  category: string;
  attempts: number;
  correct: number;
  accuracy: number;
};

export type DashboardStats = {
  total_questions: number;
  total_attempts: number;
  overall_accuracy: number;
  streak_days: number;
  due_today: number;
  daily_7d: DailyCount[];
  weak_categories: CategoryAccuracy[];
};

export type HeatmapCell = {
  category: string;
  difficulty: number;
  attempts: number;
  correct: number;
  accuracy: number;
};

export type HeatmapResponse = {
  cells: HeatmapCell[];
};

export type ProgressPoint = {
  day: string;
  attempts: number;
  correct: number;
  cumulative_attempts: number;
};

export type ProgressResponse = {
  points: ProgressPoint[];
};
